src/redux/actions/resetPassword.js

```diff
@@ -0,0 +1,119 @@
+import {
+  SAVE_INPUT,
+  PASSWORD_RESET_REQUEST,
+  PASSWORD_RESET_REQUEST_SUCCESS,
+  PASSWORD_RESET_REQUEST_FAILURE,
+  COMPLETE_RESET_REQUEST,
+  COMPLETE_RESET_REQUEST_SUCCESS,
+  COMPLETE_RESET_REQUEST_FAILURE,
+  INPUT_ERROR
+} from '../constants/resetPassword';
+import apiRequest from '../../services/apiRequest';
+import { dispatchError, dispatchSuccess, getErrorMessage } from './auth/register';
+import userValidator from '../../validators/UserValidator';
+
+const messages = {
+  404: 'User Not Found',
+  500: 'Server Error',
+  200: 'Success Message',
+  passwordUnmatch: 'Please your passwords do not match',
+  emptyInput: 'Empty Inputs',
+
+};
+
+export const saveInput = (field, value) => (dispatch) => {
+  dispatch({
+    type: SAVE_INPUT,
+    payload: {
+      field,
+      value
+    }
+  });
+};
+
+export const inputError = errors => ({
+  type: INPUT_ERROR,
+  payload: {
+    errors
+  }
+});
+export const sendResetRequest = data => (dispatch) => {
+  const errors = userValidator.validateField(data);
+  dispatch(inputError(errors));
+  if (!errors) {
+    dispatch({
+      type: PASSWORD_RESET_REQUEST,
+      payload: {
+        data,
+        loading: true
+      }
+    });
+    apiRequest
+      .startResetPassword(data)
+      .then((response) => {
+        dispatchSuccess(response.data.message, 'alert', dispatch);
+        dispatch({
+          type: PASSWORD_RESET_REQUEST_SUCCESS,
+          payload: {
+            message: response.data.message,
+            loading: false,
+
+          }
+        });
+      })
+      .catch((error) => {
+        const message = messages[error.response.status];
+        dispatchError(message, 'alert', dispatch);
+        dispatch({
+          type: PASSWORD_RESET_REQUEST_FAILURE,
+          payload: {
+            message,
+            status: 'done',
+            loading: false
+          }
+        });
+      });
+  }
+};
+
+
+export const completeResetRequest = data => (dispatch) => {
+  // const input = {password: data.password}
```

Comment:
Good job. Please remove the commented lines of code if you no longer need them.

Reply:
Thanks. I will remove the comment now.